### ISSUE TITLE
Add endpoints-sttp-client support scala.js

### ISSUE THREAD

**Issue by RustamSultanov** [julienrf/endpoints#571]
Why sttp doesnt support scala.js? Sttp have scala.js support https://sttp.softwaremill.com/en/latest/backends/javascript/fetch.html but in endpoints sttp only jvm configured :(

**Comment by julienrf** [comment]
Hello, I don’t know if there is a reason for this. Maybe at the time that client was added it didn’t support Scala.js?

in the meantime, you can use the `endpoints-xhr-client` that supports Scala.js.

**Comment by RustamSultanov** [comment]
Yes, `endpoints-xhr-client` it is good for me, but sttp support streaming via FetchMonixBackend and this is very useful in conjunction with endpoints

**Comment by julienrf** [comment]
Interesting. I don’t know how hard it would be to support Scala.js in the sttp interpreter. If it supports streaming this could solve #287. Otherwise, it should probably be possible to add support for streaming to the `endpoints-xhr-client` interpreter (there used to be a prototype here, for the record: https://github.com/julienrf/endpoints/blob/streaming-support/xhr/client/src/main/scala/endpoints/xhr/Http1Streaming.scala)
